src/seedu/addressbook/data/person/Address.java

```diff
@@ -9,12 +9,25 @@
 public class Address {
 
     public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in format of "
+    		+ "\"Block number, Street name, Unit Number, Postal Code\"";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
-
-    public final String value;
+    
+    private static final int ADDRESS_INDEX_BLOCK = 0;
+    private static final int ADDRESS_INDEX_STREET = 1;
+    private static final int ADDRESS_INDEX_UNIT = 2;
+    private static final int ADDRESS_INDEX_POSTAL = 3;
+    
+    //public final String value;
+    public final String[] addressComponents;
```

Comment:
This variable only stores the intermediate results, so it is not good to be declared as a class member here,  but the inside the method that uses it.